AQ3DC/AQ3DC.py

```diff
@@ -334,12 +335,211 @@ def saveComputationMeasuement(self):
 
             # compute all measure
             patient_compute = self.logic.computeMeasurement(self.list_measure, dict_patient)
-
+            
+            print("self.ui.ComboBoxExcelFormat.currentText : ",self.ui.ComboBoxExcelFormat.currentText)
+            if self.ui.ComboBoxExcelFormat.currentText == "Statistics":
+                patient_compute = self.reorganizeStat(patient_compute)
             # write measure
+            #test windows
             self.logic.writeMeasurementExcel(patient_compute, path, file_name)
             # self.logic.WriteMeasurementExcel2(test, path, file_name)
 
 
+    def reorganizeStat(self,patient_compute):
+        dic_stats = {
+                "ID":[],
+                "Tooth":[],
+                "Arch":[],
+                "Segment":[],
+                "Group":[],
+                "BL Ang":[],
+                "MD Ang":[],
+                "Rotation":[],
+                "AP":[],
+                "Vertical":[],
+                "Transverse-RL":[],
+                "3D":[]
+            }
+        
+        for i in range(len(patient_compute["Patient"])) :
+
+            #Tooth
+            if patient_compute["Type of measurement"][i]=="Distance between 2 points T1 T2":
+                tooths = patient_compute["Landmarks"][i]
+                indice_espace = tooths.find(" ")
+                indice_tiret = tooths.find("-")
+                if indice_espace != -1 and indice_tiret != -1:
+                    tooth = tooths[:indice_espace-1]
+
+            elif patient_compute["Type of measurement"][i]=="Angle line T1 and line T2":
+                tooth = patient_compute["Landmarks"][i][:3]
+
+            if "M" in tooth:
```

Comment:
If not all measurements are T1/T2, neither of the `if` blocks above initialize the `tooth` variable, which causes an error on this line: `UnboundLocalError: local variable 'tooth' referenced before assignment`

Are only T1/T2 measurements intended to be supported? I'd make that clearer in the UI; possibly disable the "Statistics" option if it cannot interpret the selected measurements.

Reply:
When all measurements are T1/T2 the additions seem to work as expected.

Reply:
I forgot to handle these cases thanks ! 


The statictics output is only for two teeth measurement. I've made the change, it's only visible if all the measurement can be transformed in statictics output. I also added a test to avoid the error with "'tooth' reference before assignment". 

Is it ok ? Do you see anything else to change ?